Source/Core/CustomProjection.js

```diff
@@ -0,0 +1,311 @@
+     *          coordinates are copied there and that instance is returned.  Otherwise, a new instance is
+     *          created and returned.
+     */
+    CustomProjection.prototype.project = function(cartographic, result) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!this._ready) {
+            throw new DeveloperError('CustomProjection is not loaded. Use CustomProjection.readyPromise or wait for CustomProjection.ready to be true.');
+        }
+        Check.defined('cartographic', cartographic);
+        //>>includeEnd('debug');
+
+        if (!defined(result)) {
+            result = new Cartesian3();
+        }
+
+        this._project(cartographic, result);
+        return result;
+    };
+
+    /**
+     * Unprojects a set of projected {@link Cartesian3} coordinates, in meters, to {@link Cartographic}
+     * coordinates, in radians based on the specified projection.
+     *
+     * @param {Cartesian3} cartesian The Cartesian position to unproject with height (z) in meters.
+     * @param {Cartographic} [result] An instance into which to copy the result.  If this parameter is
+     *        undefined, a new instance is created and returned.
+     * @returns {Cartographic} The unprojected coordinates.  If the result parameter is not undefined, the
+     *          coordinates are copied there and that instance is returned.  Otherwise, a new instance is
+     *          created and returned.
+     */
+    CustomProjection.prototype.unproject = function(cartesian, result) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!this._ready) {
+            throw new DeveloperError('CustomProjection is not loaded. Use CustomProjection.readyPromise or wait for CustomProjection.ready to be true.');
+        }
+        Check.defined('cartesian', cartesian);
+        //>>includeEnd('debug');
+
+        if (!defined(result)) {
+            result = new Cartographic();
+        }
+
+        this._unproject(cartesian, result);
+        return result;
+    };
+
+    function buildCustomProjection(customProjection, url) {
+        var fetch;
+        var deferred = when.defer();
+        var useCache = !isDataUri(url);
+        if (useCache && defined(CustomProjection._loadedProjectionFunctions[url])) {
+            CustomProjection._loadedProjectionFunctions[url](function(project, unproject) {
+                customProjection._project = project;
+                customProjection._unproject = unproject;
+                customProjection._ready = true;
+                deferred.resolve(customProjection);
+            });
+
+            return deferred;
+        }
+
+        if (typeof WorkerGlobalScope !== 'undefined' && self instanceof WorkerGlobalScope) { // eslint-disable-line no-undef
+            importScripts(url); // eslint-disable-line no-undef
+            fetch = when.resolve();
+        } else {
+            fetch = loadAndExecuteScript(url);
+        }
+
+        fetch = fetch
```

Comment:
What happens if the fetch fails?  There doesn't seem to be any error handling in this function.